File: FrontEnd/src/app/features/reservations/classes/dtos/form/reservation-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { CustomerActiveVM } from '../../../../customers/classes/view-models/customer-active-vm'
import { DestinationActiveVM } from '../../../../destinations/classes/view-models/destination-active-vm'
import { DriverActiveVM } from '../../../../drivers/classes/view-models/driver-active-vm'
import { PassengerReadDto } from './passenger-read-dto'
import { PickupPointActiveVM } from 'src/app/features/pickupPoints/classes/view-models/pickupPoint-active-vm'
import { PortDropdownVM } from 'src/app/features/ports/classes/view-models/port-dropdown-vm'
import { ShipDropdownVM } from 'src/app/features/ships/classes/view-models/ship-dropdown-vm'

export interface ReservationReadDto {

    reservationId: Guid
    customer: CustomerActiveVM
    destination: DestinationActiveVM
    driver: DriverActiveVM
    pickupPoint: PickupPointActiveVM
    port: PortDropdownVM
    ship: ShipDropdownVM
    date: string
    refNo: string
    email: string
    phones: string
    remarks: string
    adults: number
    kids: number
    free: number
    totalPersons: number
    ticketNo: string
    passengers: PassengerReadDto

}

File: API/Features/Reservations/Dtos/ReservationReadDto.cs
using System.Collections.Generic;
using API.Infrastructure.Classes;

namespace API.Features.Reservations {

    public class ReservationReadDto {

        public string ReservationId { get; set; }

        public string Date { get; set; }
        public string RefNo { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPersons { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public string Remarks { get; set; }
        public string UserId { get; set; }

        public SimpleResource Customer { get; set; }
        public SimpleResource Destination { get; set; }
        public PickupPointWithPortVM PickupPoint { get; set; }
        public SimpleResource Driver { get; set; }
        public SimpleResource Ship { get; set; }

        public List<PassengerReadDto> Passengers { get; set; }

    }

}